web/src/engine/websites/SenManga.ts

```diff
@@ -0,0 +1,101 @@
+import { Tags } from '../Tags';
+import icon from './SenManga.webp';
+import { Chapter, DecoratableMangaScraper, Manga, Page, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchJSON, FetchRequest, FetchWindowScript } from './../FetchProvider';
+
+const mangasPerPage = 100;
+
+type JSONManga = {
+    id: string,
+    title: string,
+    description: string,
+    language: {
+        name: string,
+        code: string
+    }
+}
+
+type JSONChapter = {
+    id: string,
+    chapter : string,
+    full_title: string
+    pages: number,
+    series : string,
+    language: {
+        name: string,
+        code: string
+    }
+}
+
+type APISingleManga = {
+    success: boolean
+    data: JSONManga
+}
+
+type APIMultiManga = {
+    success: boolean
+    data: JSONManga[]
+}
+
+type APIMultiChapter = {
+    success: boolean
+    data: JSONChapter[]
+}
+
+@Common.ImageAjax()
+
+export default class extends DecoratableMangaScraper {
+
+    public constructor() {
+        super('senmanga', `SenManga`, 'https://www.senmanga.com', Tags.Media.Manga, Tags.Language.Multilingual, Tags.Source.Aggregator);
+    }
+
+    public override async Initialize(): Promise<void> {
+        const request = new FetchRequest(this.URI.href);
+        return FetchWindowScript(request, `window.cookieStore.set('viewer', '1')`);
+    }
+
+    public override get Icon() {
+        return icon;
+    }
+
+    public override ValidateMangaURL(url: string): boolean {
+        return /^https?:\/\/www\.senmanga\.com\/title\//.test(url);
+    }
+    public override async FetchManga(provider: MangaPlugin, _url: string): Promise<Manga> {
```

Comment:
The leading underscore `_` is an informal indicator for unused parameters.
Since the parameter is used within the method i suggest to use a name without a leading underscore.